helper/predeployment/predeployment.go

```diff
@@ -0,0 +1,254 @@
+	ABI              []byte // the ABI of the Smart Contract
+	Bytecode         []byte // the raw bytecode of the Smart Contract
+	DeployedBytecode []byte // the deployed bytecode of the Smart Contract
+}
+
+// generateContractArtifact generates contract artifacts based on the
+// passed in Smart Contract JSON ABI
+func generateContractArtifact(filepath string) (*contractArtifact, error) {
+	var (
+		artifact *contractArtifact
+		err      error
+	)
+
+	// Read from the ABI from the JSON file
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		return artifact, err
+	}
+
+	jsonRaw, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return artifact, err
+	}
+
+	// Fill out the fields in the JSON file
+	var jsonResult map[string]interface{}
+	if err = json.Unmarshal(jsonRaw, &jsonResult); err != nil {
+		return artifact, err
+	}
+
+	// Parse the ABI
+	abiRaw, ok := jsonResult["abi"]
+	if !ok {
+		return nil, errABINotFound
+	}
+
+	abiBytes, err := json.Marshal(abiRaw)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal ABI to JSON, %w", err)
+	}
+
+	// Parse the bytecode
+	bytecode, ok := jsonResult["bytecode"].(string)
+	if !ok {
+		return nil, errBytecodeNotFound
+	}
+
+	hexBytecode, err := hex.DecodeString(strings.TrimPrefix(bytecode, "0x"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode bytecode, %w", err)
+	}
+
+	// Parse deployed bytecode
+	deployedBytecode, ok := jsonResult["deployedBytecode"].(string)
+	if !ok {
+		return nil, errDeployedBytecodeNotFound
+	}
+
+	hexDeployedBytecode, err := hex.DecodeString(strings.TrimPrefix(deployedBytecode, "0x"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode deployed bytecode, %w", err)
+	}
+
+	return &contractArtifact{
+		ABI:              abiBytes,
+		Bytecode:         hexBytecode,
+		DeployedBytecode: hexDeployedBytecode,
+	}, nil
+}
+
+// getArraySubstring fetches the substring of anything
+// that is between [ and ]
+func getArraySubstring(s string) string {
+	// Find the first occurrence of the opening bracket
+	i := strings.Index(s, "[")
+	if i >= 0 {
+		// Find last occurrence of the closing bracket
+		j := strings.LastIndex(s, "]")
+		if j >= 0 {
+			return s[i+1 : j]
+		}
+	}
+
+	return ""
+}
+
+// extractValue recursively extracts the values for subarrays
+// and packs them
+func extractValue(s string) interface{} {
+	if !strings.Contains(s, "[") {
+		return s
+	}
+
+	ret := make([]interface{}, 0)
+	arraySubstring := getArraySubstring(s)
+
+	if arraySubstring != "" {
+		splitValues := strings.Split(arraySubstring, ",")
+
+		for _, str := range splitValues {
+			ret = append(ret, extractValue(str))
+		}
+	}
+
+	return ret
+}
+
+// normalizeConstructorArguments cleans up the constructor arguments so that they
+// can be interpreted by the ABI encoder
+func normalizeConstructorArguments(constructorArgs []interface{}) []interface{} {
+	arguments := make([]interface{}, 0)
+
+	for _, arg := range constructorArgs {
+		argStr, _ := arg.(string)
```

Comment:
In command package, arg is casted from string to interface{}. But there is casting from interface{} to string again here. Why not normalizeConstructorArguments takes []string?

Reply:
Good catch 💯 

Previously, we didn't know how we were going to utilize the abi package in full, so we've left it as an `[]interface{}` type.

Fixed:

[b63064e](https://github.com/0xPolygon/polygon-edge/pull/536/commits/b63064e18312aa4139591c34fb8cbcf0099e8a8a)